src/ops.jl

```diff
@@ -356,3 +356,43 @@ Base.floor{T}(::Type{T}, A::FixedArray) = map(x->floor(T, x), A)
 Base.ceil{T}( ::Type{T}, A::FixedArray) = map(x->ceil( T, x), A)
 Base.trunc{T}(::Type{T}, A::FixedArray) = map(x->trunc(T, x), A)
 Base.round{T}(::Type{T}, A::FixedArray) = map(x->round(T, x), A)
+
+import Base.Cartesian.@nif
+shift(t::Tuple) = Base.tail(t)
+unshift(t::Tuple, val) = tuple(val, t...)
+push(t::Tuple, val) = tuple(t..., val)
+@generated unpush{N,T}(t::NTuple{N,T}) = begin # is there a better name?
```

Comment:
@andyferris  It's a bit of an odd story with immutable containers though: if you don't return the newly popped container the whole operation is more or less pointless.  OTOH, there's already an easy syntax to return the beginning and end.

Agreed that changing the return value isn't ideal, but `pop` and `pop!` are different methods so I doubt it's unprecedented.  The other option is just to call these something else (what though?)